driver-core/src/main/com/mongodb/operation/ChangeStreamOperation.java

```diff
@@ -0,0 +1,471 @@
+     *                     server's default value
+     * @param timeUnit     the time unit, which may not be null
+     * @return this
+     */
+    public ChangeStreamOperation<T> maxAwaitTime(final long maxAwaitTime, final TimeUnit timeUnit) {
+        notNull("timeUnit", timeUnit);
+        this.maxAwaitTimeMS = TimeUnit.MILLISECONDS.convert(maxAwaitTime, timeUnit);
+        return this;
+    }
+
+    /**
+     * Gets the maximum execution time on the server for this operation.  The default is 0, which places no limit on the execution time.
+     *
+     * @param timeUnit the time unit to return the result in
+     * @return the maximum execution time in the given time unit
+     * @mongodb.driver.manual reference/method/cursor.maxTimeMS/#cursor.maxTimeMS Max Time
+     */
+    public long getMaxTime(final TimeUnit timeUnit) {
+        notNull("timeUnit", timeUnit);
+        return timeUnit.convert(maxTimeMS, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Sets the maximum execution time on the server for this operation.
+     *
+     * @param maxTime  the max time
+     * @param timeUnit the time unit, which may not be null
+     * @return this
+     * @mongodb.driver.manual reference/method/cursor.maxTimeMS/#cursor.maxTimeMS Max Time
+     */
+    public ChangeStreamOperation<T> maxTime(final long maxTime, final TimeUnit timeUnit) {
+        notNull("timeUnit", timeUnit);
+        this.maxTimeMS = TimeUnit.MILLISECONDS.convert(maxTime, timeUnit);
+        return this;
+    }
+
+    /**
+     * Gets the read concern
+     *
+     * @return the read concern
+     * @mongodb.driver.manual reference/readConcern/ Read Concern
+     */
+    public ReadConcern getReadConcern() {
+        return readConcern;
+    }
+
+    /**
+     * Sets the read concern
+     *
+     * @param readConcern the read concern
+     * @return this
+     * @mongodb.driver.manual reference/readConcern/ Read Concern
+     */
+    public ChangeStreamOperation<T> readConcern(final ReadConcern readConcern) {
+        this.readConcern = notNull("readConcern", readConcern);
+        return this;
+    }
+
+    /**
+     * Returns the collation options
+     *
+     * @return the collation options
+     * @mongodb.driver.manual reference/command/aggregate/ Aggregation
+     */
+    public Collation getCollation() {
+        return collation;
+    }
+
+    /**
+     * Sets the collation options
+     * <p>
+     * <p>A null value represents the server default.</p>
+     *
+     * @param collation the collation options to use
+     * @return this
+     * @mongodb.driver.manual reference/command/aggregate/ Aggregation
+     */
+    public ChangeStreamOperation<T> collation(final Collation collation) {
+        this.collation = collation;
+        return this;
+    }
+
+    @Override
+    public BatchCursor<T> execute(final ReadBinding binding) {
+        if (isResumable()) {
```

Comment:
I just realized something else:  according to spec there's no need to have a retry here.  It doesn't require a retry if the initial aggregate command fails.  And it requires a single retry once the cursor has been established.  But this makes it so there are two retries.

There is a scope doc going through the process for retriable reads in general, so I think we can wait for that to go through and once it does we'll get retries of all read operations, and that will cover this case here.